socketio/src/asynchronous/client/builder.rs

```diff
@@ -97,9 +99,15 @@ impl ClientBuilder {
             max_reconnect_attempts: None,
             reconnect_delay_min: 1000,
             reconnect_delay_max: 5000,
+            data: None,
         }
     }
 
+    pub fn data<D: std::any::Any + Send + Sync>(mut self, data: Arc<D>) -> Self {
```

Comment:
Why already take the Arc? Can't we get a fat pointer? Or does that play out really badly with lifetimes?